### ISSUE TITLE
Arbitrary command line arguments crash poly scripts

### ISSUE THREAD

**Issue by eatonphil** [polyml/polyml#15]
Printing command line arguments only works if the arguments after script are a valid sml file. Here is an example file that doesn't print any arguments after the script name:

```sml
val _ = List.map PolyML.print (CommandLine.arguments())
```

Running it alone will print arguments out reasonably:

```bash
$ poly --script print.sml
"--script"
"print.sml"
```

Running it with generic arguments will cause the script not to run at all:

```bash
$ poly --script print.sml foo bar
$
```

Running it with the same script twice will not run the same script twice, but the arguments will show up more reasonably:

```bash
$ poly --script print.sml print.sml
"--script"
"print.sml"
"print.sml"
```

Running it with the name of an sml file that does not exist will also cause it to break.
```bash
$ poly --script print.sml foo.sml
$
```

Finally, this is all obviously true as well if you are using the hashbang:

```sml
#!/usr/local/bin/poly --script
val _ = List.map PolyML.print (CommandLine.arguments())
```

```bash
$ chmod +x print.sml

$ ./print.sml
"--script"
"./print.sml"

$ ./print.sml foo
$

$ ./print.sml print.sml
"--script"
"./print.sml"
"print.sml"

$ ./print.sml foo.sml
$
```

In summary, there are two odd things happening here:
1. Calling `poly --script` with multiple script names does not run each script.
2. Calling `poly --script` will throw away any arguments and crash after the script flag (perhaps there is/can be an --args flag to pass to a script?) unless they are sml files that exist.

Thanks!

**Comment by dcjm** [comment]
There was a bug where the last argument was being taken as the file name rather than the argument immediately after `--script`.  The `--script` argument is only intended for use in a shell script and so only a single instance is supported.  The `--use` argument can appear multiple times.

**Comment by eatonphil** [comment]
Thanks for looking into this. So there is ultimately no way to pass more generic arguments to the script?

**Comment by dcjm** [comment]
On 26/01/2016 18:42, Phil Eaton wrote:
> Thanks for looking into this. So there is ultimately no way to pass more generic arguments to the script?

If you are using a shell script you can pass additional arguments on the 
command line and these will be included in the arguments available to 
CommandLine.arguments().
e.g. ./myscript.ML foo bar
where myscript.ML contains
!#/usr/local/bin/poly --script
<ML code>

CommandLine.arguments() will return
["--script", "./myscript.ML", "foo", "bar"]
